File: Frontend/Logic/B20/Frontend/UiElements/Context/Impl.cs
using B20.Architecture.Contexts.Api;
using B20.Frontend.UiElements;

namespace B20.Frontend.UiElements.Context
{
    public class UiElementsImpl: ContextModule
    {
        public void Apply(ContextBuilder builder)
        {
            builder
                .SetClass<InputField>(InjectionMode.Prototype)
                .SetClass<Button>(InjectionMode.Prototype)
                .SetClass<Label>(InjectionMode.Prototype)
                .SetClass<OptionalLabel>(InjectionMode.Prototype)
                .SetClass<BoolSwitch>(InjectionMode.Prototype)
                .SetClass<EnumSwitch>(InjectionMode.Prototype);
        }
    }
}

File: Backend/lib/src/main/kotlin/com/github/bratek20/ostium/kaijugame/context/Impl.kt
package com.github.bratek20.ostium.kaijugame.context

import com.github.bratek20.architecture.context.api.ContextBuilder
import com.github.bratek20.architecture.context.api.ContextModule

import com.github.bratek20.ostium.kaijugame.api.*
import com.github.bratek20.ostium.kaijugame.impl.*

class KaijuGameImpl: ContextModule {
    override fun apply(builder: ContextBuilder) {
        builder
            .setImpl(GameApi::class.java, GameApiLogic::class.java)
    }
}